include/knowrob/queries/AnswerError.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2023, Sascha Jongebloed
+ * All rights reserved.
+ *
+ * This file is part of KnowRob, please consult
+ * https://github.com/knowrob/knowrob for license details.
+ */
+
+#ifndef KNOWROB_ANSWER_ERROR_H_
+#define KNOWROB_ANSWER_ERROR_H_
+
+#include <string>
+#include <fmt/core.h>
+#include "knowrob/terms/Term.h"
+#include "Answer.h"
+#include <knowrob/queries/ModalQuery.h>
+
+namespace knowrob {
+    /**
+     * A answer-related runtime error.
+     */
+    class AnswerError : public std::runtime_error {
```

Comment:
Why don't you use  `QueryError`?

Reply:
I thought QueryError is more specific for problems with the query or the querying. AnswerError is an error if the answer is in way that is erroenous. This is only used when parsing the answers for the ros action. If there is a modal operator this error would be thrown.